### ISSUE TITLE
CSS not working, .htaccess not affecting site after switch from nginx to apache

### ISSUE THREAD

**Issue by dme1sc** [opensourcepos/opensourcepos#3545]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code that [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable.

### Installation information

- OSPOS version is: 3.3.6
- OSPOS git commit hash is: f3e41a
- PHP version is: 7.4
- MySQL or MariaDB version is:  mariadb 10
- OS and version is:  ubuntu 20.04
- WebServer is: apache
- Selected language is: english
- (If applicable) Installation package for the LAMP/LEMP stack is: LAMP on digitalocean

### Issue / Bug  / Question / New Feature

We had a perfectly good install of ospos on nginx.  But ran into a problem with a 3rd party API integration so we decided to reconfigure to apache.  The 3rd party api integration works now.  But the major issue is that we broke the CSS mapping somehow.  This has something to do the .htaccess rewrite.  We get the ospos login and can login successfully.  The dashboard loads up but without CSS formatting - everything shows in a vertical line.  Have switched to other themes and still the same.

Confirmed /etc/apache2/apache2.conf has the override settings:
<Directory /var/www/>
        Options Indexes FollowSymLinks
        AllowOverride All
        Require all granted
</Directory>

We have edited /etc/apache2/sites-available/ospos<domain>.con.conf to show:
DocumentRoot /var/www/html/opensourcepos/public

a2enmod rewrite is enabled.

But it seems that /var/www/html/opensourcepos/public/.htaccess is ignored by the browser.  
As in Issue 920, I have tried all of the following:

RewriteRule ^(.*)$ index.php?/$1 [L]
~ systemctl restart apache2, No change

I commented out the above and uncommented and edited this line:
# if in subdir comment above line, uncomment below one and replace <OSPOS path> with your path
RewriteRule ^(.*)$ /opensourcepos/public/index.php?/$1 [L]
~ systemctl restart apache2, No change

I tried this one:
RewriteRule ^(.*)$ /var/www/html/opensourcepos/public/index.php?/$1 [L]
~ systemctl restart apache2, No change

I commented out both lines completely.
~ systemctl restart apache2, No change

So I renamed .htaccess to see what would happen.  The page loaded by just a bunch of links - no images or formatting of any kind.  I put .htaccess back in place.  So now it seems the problem is .htaccess.

This is .htaccess in its entirety:

```
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
# if in web root
RewriteRule ^(.*)$ index.php?/$1 [L]
Options +ExecCGI +Includes +IncludesNOEXEC +SymLinksIfOwnerMatch -Indexes

<IfModule mod_headers.c>
  Header always set X-Frame-Options "SAMEORIGIN"
  Header add Content-Security-Policy "default-src 'self' www.google.com; connect-src 'self'; script-src 'sHeader set X-Content-Type-Options "nosniff"
  Header set X-XSS-Protection "1; mode=block"
  Header set X-Frame-Options "DENY"
</Ifmodule>
# prevent folder listing
IndexIgnore *

# Apache 2.4
<IfModule authz_core_module>
  # secure htaccess file
  <Files .htaccess>
    Require all denied
  </Files>
  # prevent access to PHP error log
  <Files error_log>
    Require all denied
  </Files>
</IfModule>

# Apache 2.2
<IfModule !authz_core_module>
  # secure htaccess file
  <Files .htaccess>
    Order allow,deny
    Deny from all
    Satisfy all
  </Files>
  # prevent access to PHP error log
  <Files error_log>
    Order allow,deny
    Deny from all
    Satisfy all
  </Files>
</IfModule>

<IfModule mod_expires.c>
  <FilesMatch "\.(jpe?g|png|gif|js|css)$">
    ExpiresActive On
    ExpiresDefault "access plus 1 week"
  </FilesMatch>
</IfModule>
```
I'm stumped.  Thanks for reviewing.

**Comment by dme1sc** [comment]
I'm still stumped on this.

**Comment by jekkos** [comment]
Do you have mod rewrite enabled?

**Comment by jekkos** [comment]
Can you check what urls the browser is trying to load for the CSS? Where they point to?

**Comment by dme1sc** [comment]
confirmed that mod rewrite is enabled.

Console has lots of loading errors, looks like permissions.  I've reset permissions and chown to www-data.

Uncaught ReferenceError: moment is not defined
    update_clock https://apache.testxxxsites.com/home:51
    <anonymous> https://apache.testxxxsites.com/home:44
[home:51:12](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/dist/bootswatch/cerulean/bootstrap.min.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/dist/jquery-ui/jquery-ui.min.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/dist/opensourcepos.min.css?rel=88e63d8098” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/css/custom.css” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/js/qr/qrcode.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/dist/opensourcepos.min.js?rel=115cebf615” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
Loading failed for the <script> with source “https://apache.testxxxsites.com/dist/opensourcepos.min.js?rel=115cebf615”. [home:25:1](https://apache.testxxxsites.com/home)
Content Security Policy: The page’s settings blocked the loading of a resource at https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css (“style-src”).
The resource from “https://apache.testxxxsites.com/js/sweetalert2.all.min.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/js/jquery.countdown360.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/js/qr/qrcode.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
Loading failed for the <script> with source “https://apache.testxxxsites.com/js/qr/qrcode.js”. [home:28:1](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/js/jquery.countdown360.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
Loading failed for the <script> with source “https://apache.testxxxsites.com/js/jquery.countdown360.js”. [home:29:1](https://apache.testxxxsites.com/home)
The resource from “https://apache.testxxxsites.com/js/sweetalert2.all.min.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).
[home](https://apache.testxxxsites.com/home)
Loading failed for the <script> with source “https://apache.testxxxsites.com/js/sweetalert2.all.min.js”. [home:30:1](https://apache.testxxxsites.com/home)
Uncaught ReferenceError: $ is not defined
    <anonymous> https://apache.testxxxsites.com/home:54
[home:54:2](https://apache.testxxxsites.com/home)
Uncaught ReferenceError: jQuery is not defined
    <anonymous> https://apache.testxxxsites.com/home:115
[home:115:1](https://apache.testxxxsites.com/home)
Uncaught ReferenceError: dialog_support is not defined
    <anonymous> https://apache.testxxxsites.com/home:195
[home:195:2](https://apache.testxxxsites.com/home)
InstallTrigger is deprecated and will be removed in the future. [commons.js:2:1347975](moz-extension://add1cd81-1f0b-483b-b488-a65f81d03e9a/commons.js)
Content Security Policy: The page’s settings blocked the loading of a resource at data:application/font-woff2;base64,d09GM… (“font-src”).
Uncaught ReferenceError: moment is not defined
    update_clock https://apache.testxxxsites.com/home:51
    <anonymous> https://apache.testxxxsites.com/home:44
446 [home:51:12](https://apache.testxxxsites.com/home)

​

**Comment by dme1sc** [comment]
contents of ./public .htaccess

RewriteEngine On

# To redirect a subdomain to a subdir because of https not supporting wildcards
# replace values between <> with your ones
# RewriteCond %{HTTP_HOST} ^apache\.testxxxsites\.com$
# RewriteCond %{HTTP_HOST} ^www\.<OSPOS subdomain>\.<my web domain>\.com$
# RewriteRule ^/?$ "https\:\/\/apache\.testxxxsites\.com\/" [R=301,L]

# To rewrite "domain.com -> www.domain.com" uncomment the following lines.
# RewriteCond %{HTTPS} !=on
# RewriteCond %{HTTP_HOST} !^www\..+$ [NC]
# RewriteCond %{HTTP_HOST} (.+)$ [NC]
# RewriteRule ^(.*)$ http://www.%1/$1 [R=301,L]
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
# if in web root
#AddType application/octet-stream .image
#RewriteRule ^(.*)$ index.php?/$1 [L]

# if in subdir comment above line, uncomment below one and replace <OSPOS path> with your path
RewriteRule ^(.*)$ /var/www/html/opensourcepos/public/index.php?/$1 [L]

# disable directory browsing
# For security reasons, Option all cannot be overridden.
#Options All -Indexes
Options +ExecCGI +Includes +IncludesNOEXEC +SymLinksIfOwnerMatch -Indexes

<IfModule mod_headers.c>
  Header always set X-Frame-Options "SAMEORIGIN"
  Header add Content-Security-Policy "default-src 'self' www.google.com; connect-src 'self'; script-src 'self' '>
  Header set X-Content-Type-Options "nosniff"
  Header set X-XSS-Protection "1; mode=block"
  Header set X-Frame-Options "DENY"
</Ifmodule>

# prevent folder listing
IndexIgnore *

# Apache 2.4
<IfModule authz_core_module>
  # secure htaccess file
  <Files .htaccess>
    Require all denied
  </Files>
  
   # prevent access to PHP error log
  <Files error_log>
    Require all denied
  </Files>
</IfModule>

# Apache 2.2
<IfModule !authz_core_module>
  # secure htaccess file
  <Files .htaccess>
    Order allow,deny
    Deny from all
    Satisfy all
  </Files>
  # prevent access to PHP error log
  <Files error_log>
    Order allow,deny
    Deny from all
    Satisfy all
  </Files>
</IfModule>

<IfModule mod_expires.c>
  <FilesMatch "\.(jpe?g|png|gif|js|css)$">
    ExpiresActive On
    ExpiresDefault "access plus 1 week"
  </FilesMatch>
</IfModule>

**Comment by dme1sc** [comment]
When I go to these links in the console errors - it loads the css page as text....
example:  https://apache.testxxxsites.com/css/custom.css

.modal-body #qrcode img{ margin: 0px auto; display: table !important; } html { overflow-x: hidden !important; } #qrCodeModal .modal-header .close { position: absolute; top: 8px; right: 15px; color: black; opacity: 1; } #qrCodeModal .modal-header .close span { color: black; font-size: 28px; } #qrCodeModal .modal-header #exampleModalLabel { font-size: 24px; text-align: center; } div#qrcode canvas{ margin: 0px auto; display: table; } #customer_form #customer_basic_info .radio-inline + .radio-inline{ margin-left: 8px; } #register_wrapper #mode_form .panel-body ul .pull-left { margin-left: 12px; } .new_layout .col-xs-4.left_info_item { width: 35%; } .new_layout .col-xs-5.center_info_item { width: 45%; padding: 0; } .new_layout .col-xs-3.right_info_item { width: 20%; } .new_layout .col-xs-4.left_info_item #items_register_wrapper { width: 100%; float: left; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 { padding: 0; position: relative; border: 1px solid #2ea3e6; width: 32%; margin: 0 0 2% 2%; position: relative; overflow: hidden; cursor: pointer; float: left; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4:nth-child(3n+1) { margin-left: 0; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4::before { content: ''; background: rgba(2, 52, 99, 0.4); height: 100%; width: 100%; position: absolute; z-index: 9; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 img { height: 185px; object-fit: contain; width: 100%; transition: 0.5s all ease-in-out; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4:hover img { transform: scale(1.2); } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 .right-info { float: left; width: 100%; padding: 10px 10px; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 .right-info span { float: left; width: 100%; font-size: 16px; font-weight: 600; color: #fff; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 h2 { float: left; width: 60%; margin: 0; font-size: 16px; color: #fff; min-height: 50px; font-weight: 600; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 .item-des_info { position: absolute; top: 50%; transform: translateY(-50%); padding: 0 5px; display: flex; align-items: center; z-index: 99; justify-content: space-between; width: 100%; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4 .right-info { float: right; width: 40%; padding: 0 0 0 4px; text-align: right; } #register_wrapper #mode_form .panel-body ul .pull-left { display: grid; text-align: left; } .new_layout .col-xs-4.left_info_item #items_register_wrapper .col-md-12 .col-md-4:hover::before { background: rgba(2, 52, 99, 0.6); background: rgba(0, 0, 0, 0.6); } #register_wrapper #mode_form .panel-body ul .pull-left .control-label { text-align: left; color: #000; margin-bottom: 2px; } #register_wrapper #mode_form .panel-body ul .pull-left input { height: 32px; width: auto; max-width: 100px; } div#overall_sale .form-group.form-group-sm .col-xs-6 { width: 100%; padding: 0; } #register_wrapper #mode_form .panel-body ul li.pull-right { margin-top: 3

**Comment by jekkos** [comment]
Can you try to remove the line with nosniff in .htaccess?

**Comment by jekkos** [comment]
This one specifically 

```
  Header set X-Content-Type-Options "nosniff"
```

**Comment by dme1sc** [comment]
Commented out and restarted apache2, no change.

**Comment by jekkos** [comment]
I think the issue is that it's returning the wrong mimetype. I thought the header would cause that but that seems to be not the case.

**Comment by jekkos** [comment]
Can you check if mod_mime is enabled and configured correctly for css and js?

**Comment by dme1sc** [comment]
Hi, your last note may be dig into the apache2.conf file looking for the mod_mime settings for css and javascript.  I played around with it and it didn't help.  But I noticed at the end of the file a line of:
SetHandler application/x-httpd-php

I commented this out and now it works.  Go figure.   

Thanks jekkos for the deep-dive here.  I know it was more than ospos related.  I'll send a tip to the the wallet.
